modules/gapi/perf/common/gapi_render_perf_tests.cpp

```diff
@@ -0,0 +1,1281 @@
+                                Values(LINE_8),
+                                Values(0),
+                                Values(CV_8UC3)));
+
+INSTANTIATE_TEST_CASE_P(RenderTestLines, RenderTestLines,
+                        Combine(Values(szVGA, sz720p, sz1080p),
+                                Values(cv::Point(100, 100)),
+                                Values(cv::Point(200, 200)),
+                                Values(cv::Scalar(100, 50, 150)),
+                                Values(2),
+                                Values(LINE_8),
+                                Values(0),
+                                Values(CV_8UC3)));
+
+INSTANTIATE_TEST_CASE_P(RenderTestMosaics, RenderTestMosaics,
+                        Combine(Values(szVGA, sz720p, sz1080p),
+                                Values(cv::Rect(100, 100, 200, 200)),
+                                Values(25),
+                                Values(0),
+                                Values(CV_8UC3)));
+
+INSTANTIATE_TEST_CASE_P(RenderTestImages, RenderTestImages,
+                        Combine(Values(szVGA, sz720p, sz1080p),
+                                Values(cv::Rect(50, 50, 100, 100)),
+                                Values(cv::Scalar(100, 150, 60)),
+                                Values(1.0),
+                                Values(CV_8UC3)));
+
+INSTANTIATE_TEST_CASE_P(RenderTestPolylines, RenderTestPolylines,
+                        Combine(Values(szVGA, sz720p, sz1080p),
+                                Values(std::vector<cv::Point>{{100, 100}, {200, 200}, {150, 300}, {400, 150}}),
+                                Values(cv::Scalar(100, 150, 60)),
+                                Values(2),
+                                Values(LINE_8),
+                                Values(0),
+                                Values(CV_8UC3)));
+
+INSTANTIATE_TEST_CASE_P(RenderTestPolyItems, RenderTestPolyItems,
+                        Combine(Values(szVGA, sz720p, sz1080p),
+                                Values(50),
+                                Values(50),
+                                Values(50)));
+
+
+#ifdef HAVE_FREETYPE
+PERF_TEST_P_(RenderTestFTexts, RenderFTextsPerformanceBGROCVTest)
+{
+    std::wstring text;
+    cv::Size sz;
+    cv::Point org;
+    int fh;
+    cv::Scalar color;
+    MatType type;
+
+    std::tie(text ,sz ,org ,fh ,color, type) = GetParam();
+
+    cv::Mat gapi_mat, ref_mat;
+    create_rand_mats(sz, type, ref_mat, gapi_mat);
+
+    // G-API code //////////////////////////////////////////////////////////////
+    cv::gapi::wip::draw::Prims prims;
+    prims.emplace_back(cv::gapi::wip::draw::FText{text, org, fh, color});
+
+    cv::GMat in;
+    cv::GArray<cv::gapi::wip::draw::Prim> arr;
+
+    cv::GComputation comp(cv::GIn(in, arr),
+                          cv::GOut(cv::gapi::wip::draw::render3ch(in, arr)));
+
+    // Warm-up graph engine:
+    comp.apply(gin(gapi_mat, prims), gout(gapi_mat));
```

Comment:
The problem with this tests now is that the tests can't be instantiated with another backend. Please follow the general model of having `compile_args` as one of test's parameters.

We need to reuse these tests on other platforms.

Also, since we need to do this, this file needs to be a header file not a .cpp, right?

Have a look on this: https://github.com/opencv/opencv/blob/3a46321fef901b8635cf5040a1f1764ffeb86db7/modules/gapi/perf/common/gapi_core_perf_tests.hpp

...And this: https://github.com/opencv/opencv/blob/3a46321fef901b8635cf5040a1f1764ffeb86db7/modules/gapi/perf/common/gapi_core_perf_tests_inl.hpp

Reply:
@dmatveev Let me explain, in case `render` we implement tests for every backend separately, because there is no chance to implement the same reference versions

Reply:
@TolyaTalamanov then it shouldn't be common.